### PR TITLE
[fix]: comma misbehave in profile form section

### LINKED ISSUE #8040: [BUG] comma's misbehave in the form, editor tags section
### Description

Comma at the end of the tags should be ignored 
image

Also, with a comma at the end make one of the central comma's a double comma ",," will give you an empty tag in the middle.  When you then delete one of the extra comma's the empty tag never goes away.
image

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Ashish-simpleCoder**
## Fixes Issue

Closes #8040 

## Changes proposed
- Empty blocks in the tag list won't appear anymore.
- Updating the tags with the response of the profile-update api.
- Updated the key prop's value to the ```index```, if same-named tags are entered multiple times and also for removing them when user removes them from the tag-input field section. Rendered ```Tag``` Components will not unmount if new tags are added.

- Backend API Changes -> Added filter for removing all of the falsy value and string values which contains only whitespaces to prevent whitespaced link generation for the tags when user clicks on them.

Example
if input is ```" javascript,          nextjs        ,,,,,solidjs, vuejs,"```, then they will be result into the ```"javascript, nextjs,solidjs,vuejs"``` while saving in db.

## Check List (Check all the applicable boxes) 

- [X] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] All new and existing tests passed.
- [X] This PR does not contain plagiarized content.
- [X] The title of my pull request is a short description of the requested changes.

## Screenshots

This is the screenshot of the tags before saving them.
before_save_tags

This is the screenshot after saving the profile changes.
after_save_tags

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/8087"></a>

### CODE REVIEW

File: components/user/UserProfile.js [hunk 1]
```diff
@@ -76,15 +76,19 @@ function UserProfile({ BASE_URL, data }) {
...
-              <Tag
-                name={tag}
-                key={tag.toLowerCase()}
-                onClick={() =>
-                  router.push(`/search?keyword=${tag.toLowerCase()}`)
-                }
-              />
-            ))}
+            data.tags.map((tag, index) => {
+              const trimmed_tag = tag.trim();
```
Comment (eddiejaoude) [inline components/user/UserProfile.js:80]: the naming does not follow the project standards, it should be `trimmedTag`

File: components/user/UserProfile.js [hunk 2]
```diff
@@ -76,15 +76,19 @@ function UserProfile({ BASE_URL, data }) {
...
+              if (!trimmed_tag) return null;
```
Comment (eddiejaoude) [inline components/user/UserProfile.js:81]: this style does not follow the project standard, we use `{}` for conditions

File: components/user/UserProfile.js [hunk 3]
```diff
@@ -76,15 +76,19 @@ function UserProfile({ BASE_URL, data }) {
...
-            data.tags.map((tag) => (
-              <Tag
-                name={tag}
-                key={tag.toLowerCase()}
-                onClick={() =>
-                  router.push(`/search?keyword=${tag.toLowerCase()}`)
-                }
-              />
-            ))}
+            data.tags.map((tag, index) => {
```
Comment (eddiejaoude) [inline components/user/UserProfile.js:79]: Maybe a `flatMap` would be better here, so items can be removed?

File: pages/account/manage/profile.js [hunk 4]
```diff
@@ -93,6 +93,7 @@ export default function Profile({ BASE_URL, profile, fileExists }) {
         ).join(", ")}`,
       });
     }
+    setTags(update.tags);
```
Comment (eddiejaoude) [inline pages/account/manage/profile.js:96]: good spot 👍

File: pages/api/account/manage/profile.js [hunk 5]
```diff
@@ -57,7 +57,9 @@ export async function updateProfileApi(username, data) {
     layout: data.layout,
     name: data.name,
     bio: data.bio,
-    tags: data.tags.filter((tag) => tag !== ""),
+    tags: data.tags
+      .filter((tag) => Boolean(tag.trim()))
```
Comment (eddiejaoude) [inline pages/api/account/manage/profile.js:61]: Not sure what this is for?

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Thank you 👍 I left some inline comments

File: components/user/UserProfile.js (hunk 1 above)
Comment (Ashish-simpleCoder) [reply components/user/UserProfile.js]: I am sorry. I will change this.

File: components/user/UserProfile.js (hunk 3 above)
Comment (Ashish-simpleCoder) [reply components/user/UserProfile.js:79]: There is no need for flatting the array , we only need to remove falsy values.
We can also filter the items before mapping , but that will result into twice iteration over the tags list.
That's why I have choosen that approach.

File: pages/api/account/manage/profile.js (hunk 5 above)
Comment (Ashish-simpleCoder) [reply pages/api/account/manage/profile.js:61]: Are you talking about Array.filter(item => Boolean(item))?
This is filter technique, which can filter out any falsy values from an array. Values like null, undefined, false , "" etc which are falsy in Javascript, they will be removed from array.

Comment (Ashish-simpleCoder) [reply pages/api/account/manage/profile.js:61]: Prevoius code was not handling cases for strings like ```"     "```, which contains whitespace.

Comment (eddiejaoude) [reply pages/api/account/manage/profile.js:61]: ok thank you! Please don't resolve in the future, otherwise I might miss your replies

**Review by eddiejaoude — APPROVED** [review]
Looks good 👍  thank you!

**Pull request merged**